src/openvic-simulation/economy/trading/MarketInstance.cpp

```diff
@@ -0,0 +1,28 @@
+#include "MarketInstance.hpp"
+
+using namespace OpenVic;
+
+bool MarketInstance::setup(GoodInstanceManager const& new_good_instance_manager) {
+	good_instance_manager = &new_good_instance_manager;
+	return true;
+}
+
+SellResult MarketInstance::sell(GoodDefinition const& good, const fixed_point_t quantity) {
+	GoodInstance const* const good_instance = good_instance_manager->get_good_instance_by_identifier(good.get_identifier());
```

Comment:
I'm assuming setup was called before sell.
If desired, I can add a null check + log.

Reply:
Setup is replaced by constructor and we use references now instead of pointers so this no longer matters.